### ISSUE TITLE
Support layout templates for all page kinds

### ISSUE THREAD

**Issue by akbyrd** [gohugoio/hugo#12988]
Use case: I want to make a single layout template that applies to both section and page kinds.

I have a custom output format for Atom feeds and a template to render them.

Currently I need to create both `layouts/_default/list.atom.xml` and `layouts/_default/single.atom.xml`. However, as far as I can tell RSS is hard coded with a special case for `layouts/_default/rss.xml` where the page kind is not required. This is great - I propose the same behavior for all layout templates.

If I try to write the valid template locations for home and section page kinds it looks something like this:

home
```
layouts[/{<type>|_default}]/{<layout>|index|<kind>|list}.<format>
optional for RSS            ^---------------------------^
```

section
```
layouts/{<type>|<section>|<kind>|/_default}/{<layout>|<section>|<kind>|list}.<format>
optional for RSS                            ^-------------------------------^
```

Where `{a|b}` means one of several values, `[]` means optional, and `<>` means a value replaced by page data. Not 100% accurate and I've left out a lot of detail, but I think close enough for this discussion.

For the part I've marked as optional for RSS it would be great if it were optional for other layout templates as well.

This is added functionality, so is less likely to break existing sites. It also seems like it might make the template locations slightly easier to understand since it's more consistent.

**Comment by jmooring** [comment]
> layouts/_default/rss.xml where the page kind is not required

`rss` is both a page kind and an output format, so we have a lookup order that is inconsistent and perhaps confusing when compared to other output formats. It sounds like this is a legacy thing that we continue to support for backwards compatibility. See  <https://github.com/gohugoio/hugo/issues/12992#issuecomment-2445351908>.

If I could wave a magic wand and somehow not break existing sites, `rss` would not be a page kind, and the lookup order for the `rss` output format would be consistent with everything else. 

I understand the desire to have one template handle all page kinds, but the purpose of a template named `html.html` is, at least to me, unclear. Which is why I don't like `rss.xml` either.

If the `baseof` template could function without the base/block construct, something like `baseof.atom.xml` makes sense to me, but I have no idea if that would even be possible given the different context passed to each template time (e.g. `Page.Pages`, `Page.Data.Term`, etc.).

**Comment by bep** [comment]
>I understand the desire to have one template handle all page kinds, but the purpose of a template named html.html is, at least to me, unclear. Which is why I don't like rss.xml either.

The main purpose is, as I see it, to preserve the file extension to make the file editor happy. A common setup would be to have multiple output formats with the same output format (e.g. amp.html and ... html.html, search.json data.json ...) but different templates. Of course, `html.html` looks odd, but that's how it is.

>If the baseof template could function without the base/block construct, something like baseof.atom.xml 

Baseof templates does not work without the base/block construct, but baseof.atom.xml works fine (base/block could work nicely with base/block, I guess).

**Comment by bep** [comment]
>rss is both a page kind and an output format,

I think it's safe to say that `rss` _was_ a page kind at some point.

**Comment by jmooring** [comment]
Well, this still works...

```text
disableKinds = ['rss']
```

So it sure _looks_ like a page kind, even if it isn't under the hood.

**Comment by bep** [comment]
>So it sure looks like a page kind, even if it isn't under the hood.

No, it isn't. See https://gohugo.io/getting-started/glossary/#page-kind

We have code in place to make old sites work, which I think I commented about in #12992.
